### PR TITLE
INFENG-6103 allow no origin header requests

### PR DESCRIPTION
**Pull request by blakeroberts-wk**
https://jira.atl.workiva.net/browse/INFENG-6103

## What
- Allow requests w/o origin header to pass
- Add Dockerfile
- Add Glide
- Update Travis

@Workiva/product2

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(1) Security relevant changes were detected</summary>
<li>Watched file <code>rest/middleware/cors.go</code> modified</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+RUN git config --global url.git@github.com:.insteadOf https://github.com/
+RUN mkdir ~/.ssh; ssh-keyscan -t rsa github.com > ~/.ssh/known_hosts
+RUN chmod -R 700 ~/.ssh; echo "${GIT_SSH_KEY}" > ~/.ssh/id_rsa; chmod 600 ~/.ssh/id_rsa
+RUN eval "$(ssh-agent -s)" && ssh-add ~/.ssh/id_rsa
+
+WORKDIR /go/src/github.com/Workiva/go-rest
+COPY . /go/src/github.com/Workiva/go-rest
+
+#install glide and install dependencies
+RUN go get -u github.com/Masterminds/glide
```
Comment (brianshannan-wf) [inline Dockerfile]: nit: I'd add this before the repo copy so it could potentially be cached

File: glide.lock [hunk 2]
```diff
@@ -0,0 +1,31 @@
+hash: aa3313d279fdff7629632676678092f79f7d06aa660ea28595f6d1fe79c1a14f
```
Comment (brianshannan-wf) [inline glide.lock]: As this is a library we shouldn't check in the lock file

File: glide.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package: github.com/Workiva/go-rest
+import:
+- package: github.com/gorilla/context
+  version: ~1.1.1
+- package: github.com/gorilla/mux
+  version: ~1.6.2
+- package: github.com/blakeroberts-wk/mustache
```
Comment (brianshannan-wf) [inline glide.yaml]: I'm not sure what the rules around pulling from our forks are, we should look into that

File: rest/middleware/cors.go [hunk 4]
```diff
@@ -13,25 +13,32 @@ import (
 // a MiddlewareError if the request should be terminated.
 func NewCORSMiddleware(originWhitelist []string) rest.Middleware {
 	return func(w http.ResponseWriter, r *http.Request) *rest.MiddlewareError {
+		origin := r.Header.Get("Origin")
 		originMatch := false
-		if origin := r.Header.Get("Origin"); checkOrigin(origin, originWhitelist) {
+		if checkOrigin(origin, originWhitelist) {
```
Comment (brianshannan-wf) [inline rest/middleware/cors.go]: Should we add a check for the origin header being missing here, as opposed to doing the check unnecessarily?

File: glide.yaml (hunk 3 above)
Comment (blakeroberts-wk) [reply glide.yaml]: Waiting on a Workiva fork: https://jira.atl.workiva.net/browse/COREAPPS-2112

**Comment by matthewsullivan-wf** [comment]
+1 security, all appears to be well

**Comment by brianshannan-wf** [comment]
+1

**Comment by blakeroberts-wk** [comment]
@Workiva/release-management-p

**Pull request merged**